### DISCUSSION TITLE
Option to define a player's AttackFrame duration regardless of weapon speed

### DISCUSSION THREAD

**Discussion by Mcadams** [AscensionGameDev/Intersect-Engine#2008] [Ideas]
Right now if you use attack frames the frames will play out at a speed of Player's Attack Speed divided by # of frames. What this does it makes weapons that aren't absurdly fast look and feel clunky to use. For example, this large sword is set to be able to be swung every 3 seconds (3000ms) and looks like this:

https://i.imgur.com/gQSZX8t.gif

I'd like the option to be able to set the player's AttackFrame durations, regardless of their weapon speed, so attack animations will always be smooth regardless of how often the player can attack. Below is the weapon at .4s speed (400ms), which looks great, but causes the player to attack far too quickly

https://i.imgur.com/jFfkKHd.gif

**Comment by da-rulez** [comment]
Agree!

**Comment by PyroTech03** [comment]
Solution would essentially to behave like a cooldown. Play the normal animation for the attack then back load the remainder of the attack time.
